python-checks/src/test/resources/checks/sillyIdentity.py

```diff
@@ -18,3 +18,42 @@ def foo():
     if a is a2: # OK
         pass
 
+def numeric_literals():
+    a = A()
+    if a is 42: # Noncompliant
+        pass
+
+    if 42.0 is 42: # Noncompliant
+        pass
+
+    if 42 is 42: # OK
+        pass
+
+    if 42.0 is 42.0: # OK
+        pass
+
+
+def string_literals():
+    a = A()
+    if a is "foo": # Noncompliant
+        pass
+
+    if b"foo" is "foo": # Noncompliant
+        pass
+
+    if "foo" is "foo": # OK
+        pass
+
+    if 42 is "42": # Noncompliant
+        pass
+
+def other_literals():
+    a = A()
+    if a is []: # Noncompliant
+        pass
+
+    if [] is []: # FN
+        pass
+
+    if [] is {}: # Noncompliant
+        pass
```

Comment:
I don't really see the value of those test cases.

Reply:
Ok, I will keep a couple of them which I believe have some values, like documented false negative at line 55.